### PR TITLE
Changes to support views in hive catalogs

### PR DESCRIPTION
**Pull request by zhljen**

### CODE REVIEW

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/model/TableInfo.java [hunk 1]
```diff
@@ -39,6 +39,9 @@
 public class TableInfo extends BaseInfo {
     private List<FieldInfo> fields;
     private StorageInfo serde;
+    private String viewOriginalText;
```
Comment (ajoymajumdar) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/model/TableInfo.java]: Add comment saying that the below fields are used when a table is of type 'TABLE_VIEW'.

Comment (ajoymajumdar) [reply metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/model/TableInfo.java]: Also I think we should put these three fields in a separate class. Either we have ViewInfo that extends TableInfo or be a member of TableInfo.

File: metacat-common/src/main/java/com/netflix/metacat/common/dto/TableDto.java [hunk 2]
```diff
@@ -68,6 +68,12 @@
     private QualifiedName name;
     @ApiModelProperty(value = "serialization/deserialization info about the table")
     private StorageDto serde;
+    @JsonProperty
```
Comment (ajoymajumdar) [inline metacat-common/src/main/java/com/netflix/metacat/common/dto/TableDto.java]: Put these in a separate class. Also add swagger docs.

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorTableService.java [hunk 3]
```diff
@@ -233,6 +241,30 @@ void updateTable(
...
+                    table.getDbName(),
+                    table.getDbName()));
+        }
+
+        if (Strings.isNullOrEmpty(table.getViewExpandedText())) {
+            //set viewExpandedText to viewOriginalTest
+            table.setViewExpandedText(table.getViewOriginalText());
+        }
+        //setting dummy string to view to avoid dropping view issue in hadoop Path org.apache.hadoop.fs
+        if (Strings.isNullOrEmpty(table.getSd().getLocation())) {
```
Comment (ajoymajumdar) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorTableService.java]: Did we try setting this to null?

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/model/TableInfo.java (hunk 1 above)
Comment (ajoymajumdar) [reply metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/model/TableInfo.java]: Also can we get rid of the table type. If viewOriginalText is set, then assume that it is a view.

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/converter/ConverterUtil.java [hunk 4]
```diff
@@ -81,6 +83,7 @@ protected void configure() {
                 mapping(PartitionDto.class, PartitionInfo.class)
                     .fields("name", "name", FieldsMappingOptions.copyByReference());
                 mapping(AuditDto.class, AuditInfo.class);
+                mapping(HiveVirtualViewDto.class, HiveVirtualViewInfo.class);
                 mapping(StorageDto.class, StorageInfo.class);
```
Comment (zhljen) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/converter/ConverterUtil.java]: Somehow this mapping isn't working in the TableDto <-> TableInfo. Added a get around.

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/model/HiveVirtualViewInfo.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ * @author zhenl
+ * @since 1.2.0
+ */
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+public class HiveVirtualViewInfo {
```
Comment (ajoymajumdar) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/model/HiveVirtualViewInfo.java]: Lets just name it as ViewInfo

File: metacat-common/src/main/java/com/netflix/metacat/common/dto/HiveVirtualViewDto.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ * @author zhenl
+ * @since 1.2.0
+ */
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+public class HiveVirtualViewDto extends BaseDto {
```
Comment (ajoymajumdar) [inline metacat-common/src/main/java/com/netflix/metacat/common/dto/HiveVirtualViewDto.java]: Rename it to ViewDto

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/converter/ConverterUtil.java [hunk 7]
```diff
@@ -137,7 +140,8 @@ public TableDto toTableDto(final TableInfo tableInfo) {
      * @return connector table info
      */
     public TableInfo fromTableDto(final TableDto tableDto) {
-        return mapper.map(tableDto, TableInfo.class);
```
Comment (ajoymajumdar) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/converter/ConverterUtil.java]: Replace it with the one line return.

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/converter/ConverterUtil.java [hunk 8]
```diff
@@ -150,6 +154,27 @@ public PartitionDto toPartitionDto(final PartitionInfo partitionInfo) {
         return mapper.map(partitionInfo, PartitionDto.class);
     }
 
+    /**
```
Comment (ajoymajumdar) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/converter/ConverterUtil.java]: Do we need the two methods below?

**Comment by zhljen** [comment]
Added changes to address comments.

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/converter/ConverterUtil.java (hunk 7 above)
Comment (zhljen) [reply metacat-common-server/src/main/java/com/netflix/metacat/common/server/converter/ConverterUtil.java]: Changed.

**Pull request merged**